documentation/book/oauth/con-oauth-authentication-client-options.adoc

```diff
@@ -0,0 +1,79 @@
+* An {oauth} introspection endpoint provided by authorization server
+
+Using fast local token validation requires the authorization server to provide a JWKS endpoint with public certificates that are used to validate signatures on the tokens.
+
+NOTE: An authorization server might only allow the use of opaque access tokens, which means that local token validation is not possible.
+
+Another option is to use an {oauth} introspection endpoint on the authorization server.
+Each time a Kafka broker connection is established, the broker sends the access token it receives to the authorization server, and a response confirming whether or not the token is valid is returned.
+
+Kafka client credentials can also be configured for:
+
+* Direct local access using a previously generated long-lived access token
+* Contact with the authorization server for a new access token to be issued and sent to the Kafka broker
+
+== Example client authentication flows
+
+Here you can see example flows using the following {oauth} client authentication options.
+
+* xref:oauth-introspection-endpoint[Authorization server authentication using an introspection endpoint]
+* xref:oauth-jwt[Authorization server authentication using a local JWT token signature validation]
+* xref:oauth-token-endpoint[Authentication using a long-lived access token and introspection endpoint]
+* xref:oauth-token-jwt[Authentication using a long-lived token and a local JWT token signature validation]
+
+[id='oauth-introspection-endpoint']
+.Authorization server authentication using an introspection endpoint
+
+image:oauth-introspection-endpoint.png[Authorization server authentication using an introspection endpoint]
+
+. Kafka client authenticates with the authorization server from the token endpoint using a client ID and secret.
+. An access token is returned by the authorization server.
+. Kafka client authenticates with the Kafka broker using the _SASL OAUTHBEARER_ mechanism to pass the access token.
+. Kafka broker validates the access token by calling a token introspection endpoint provided by the authorization server using a client ID and secret.
+. Kafka client session is established if the token is valid.
+
+[id='oauth-jwt']
+.Authorization server authentication using a local JWT token signature validation
+
+image:oauth-jwt-signature.png[Authorization server authentication using a local JWT token signature]
+
+. Kafka client authenticates with the authorization server from the token endpoint using a client ID, an authentication secret and a refresh token.
+. An access token is returned by the authorization server.
+. Kafka client authenticates with the Kafka broker using the _SASL OAUTHBEARER_ mechanism to pass the access token.
+. Kafka broker validates the access token locally using a JWT token signature and the Kafka client session is established.
+
+[id='oauth-token-endpoint']
+.Authentication using a long-lived access token and introspection endpoint
```

Comment:
.Client using a long-lived access token / Broker delegating validation to authorization server